### PR TITLE
➖ DEV: Remove Python 3.7 Support

### PR DESCRIPTION
**Pull request by shaneahmed**
- Remove Python 3.7 support

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -64,10 +64,10 @@ repos:
     hooks:
       - id: isort
   - repo: https://github.com/psf/black
-    rev: 22.10.0 # Replace with any tag/version: https://github.com/psf/black/tags
+    rev: 22.12.0 # Replace with any tag/version: https://github.com/psf/black/tags
```
Comment (John-P) [inline .pre-commit-config.yaml:67]: Should probably not be in this PR due to being unrelated to removing Python 3.7

File: HISTORY.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # History
 
-# 1.3.1 (2022-12-20)
+## 1.3.1 (2022-12-20)
```
Comment (John-P) [inline HISTORY.md:4]: Should probably be a separate PR for clean up as unrelated to 3.7

File: docs/requirements.txt [hunk 3]
```diff
@@ -1,7 +1,6 @@
-furo>=2022.6.21
-jinja2>=3.0.3, <3.1.0
+furo>=2022.12.7
 myst-nb>=0.17.1
 myst-parser>=0.18.1
-sphinx-copybutton>=0.5.0
-sphinx-toolbox>=3.1.2
+sphinx-copybutton>=0.5.1
+sphinx-toolbox>=3.2.0
```
Comment (John-P) [inline docs/requirements.txt:5]: Bumping versions should probably be a separate PR as unrelated to dropping 3.7

File: requirements.txt [hunk 4]
```diff
@@ -4,28 +4,28 @@ albumentations>=1.3.0
...
+sphinx>=5.3.0
+tifffile>=2022.10.10
+torch>=1.13.1
+torchvision>=0.14.1
 tqdm>=4.64.1
 umap-learn>=0.5.3
-wsidicom>=0.3.2
-zarr>=2.12.0
+wsidicom>=0.5.0
+zarr>=2.13.3
```
Comment (John-P) [inline requirements.txt:31]: Changing pinned versions should probably be a separate PR as unrelated to dropping 3.7

**Review by John-P — APPROVED** [review]
Looks good but there are a few changed such as formatting fixes and changing pinned package versions which are independent of dropping Python 3.7 and should probably really be a separate PR. However, I'm sure theses changes would also be merged right after if the tests pass so I'm ok with that.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (shaneahmed) [reply .pre-commit-config.yaml]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/526/commits/3e370499a7edc12328fb86034c0dbe8be725b9d4

File: docs/requirements.txt (hunk 3 above)
Comment (shaneahmed) [reply docs/requirements.txt]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/526/commits/3e370499a7edc12328fb86034c0dbe8be725b9d4

File: requirements.txt (hunk 4 above)
Comment (shaneahmed) [reply requirements.txt]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/526/commits/3e370499a7edc12328fb86034c0dbe8be725b9d4

File: HISTORY.md (hunk 2 above)
Comment (shaneahmed) [reply HISTORY.md:4]: I will update dependencies in a separate PR but keeping this change in this PR as this is one line only.
